test/test_ops.py

```diff
@@ -203,56 +209,66 @@ class TestCommon(JitCommonTestCase):
     #   against eager's gold standard op function variant
     @_variant_ops(op_db)
     def test_variant_consistency_eager(self, device, dtype, op):
-        samples = op.sample_inputs(device, dtype, requires_grad=op.supports_autograd)
+        # Acquires variants (method variant, inplace variant, aliases)
+        method = op.get_method()
+        inplace = op.get_inplace()
 
-        for sample in samples:
-            # Acquires variants (method variant, inplace variant, aliases)
-            method = op.get_method()
-            inplace = op.get_inplace()
+        # list of all inplace ops: inplace variant + alias inplace variants if exist
+        inplace_ops = [inplace, ]
 
-            # list of all inplace ops: inplace variant + alias inplace variants if exist
-            inplace_ops = [inplace, ]
-
-            aliases = []
-            for a_op in op.aliases:
-                aliases.append(a_op.op)
-                aliases.append(a_op.method_variant)
-                aliases.append(a_op.inplace_variant)
-                inplace_ops.append(a_op.inplace_variant)
-            aliases = tuple(aliases)
-
-            inplace_ops = tuple(v for v in inplace_ops if v is not None)
-            variants = (v for v in (method, inplace) + aliases if v is not None)
-
-            # Computes function forward and backward values
-            sample.input.grad = None
-            expected_forward = op(sample.input, *sample.args, **sample.kwargs)
-            expected_grad = None
-
-            # TODO: backward consistency only supported for single tensor outputs
-            # TODO: backward consistency only checked on sample.input, not all
-            #   tensor inputs
-            # TODO: update to handle checking grads of all tensor inputs as
-            #   derived from each tensor output
-            if (op.supports_autograd and isinstance(expected_forward, torch.Tensor)):
-                expected_forward.sum().backward()
-                expected_grad = sample.input.grad
-
-            # Test eager consistency
-            for variant in variants:
-                # Compares variant's forward
-                # Note: copies the to-be-modified input when testing the inplace variant
+        aliases = []
+        for a_op in op.aliases:
+            aliases.append(a_op.op)
+            aliases.append(a_op.method_variant)
+            aliases.append(a_op.inplace_variant)
+            inplace_ops.append(a_op.inplace_variant)
+        aliases = tuple(aliases)
+
+        inplace_variants = tuple(v for v in inplace_ops if v is not None)
+        variants = tuple(v for v in (method, inplace) + aliases if v is not None)
+        outplace_variants = tuple(set(variants) - set(inplace_ops))
+
+        samples = op.sample_inputs(device, dtype, requires_grad=op.supports_autograd,
+                                   for_inplace_variant=False)
+
+        def _test_consistency_helper(samples, op_variants):
+            for sample in samples:
```

Comment:
Reason for iterating over samples and then over variants (instead of taking product of samples and variants and calling the helper) is so that we can utilise the `expected_forward` and `expected_grad` for multiple variants (from the same variant group).

outplace_variants will include method, alias function, alias method (if available)
inplace_variants will include inplace method, and alias inplace_method  (if available)

Reply:
You're right, this is a good refinement of the design.

Reply:
This explains the reason for previous test following this structure :)

Reply:
Yep, although the acquisition of the variants can still be done once upfront I think, right?

Reply:
Yes.